defender/src/auto-tasks/batch-mint-claims-from-allowlists.ts

```diff
@@ -59,12 +59,16 @@ export async function handler(event: AutotaskEvent) {
   );
   console.log("Formatted claim ids", formattedClaimIds);
 
-  const deleteResult = await client
-    .from(network.supabaseTableName)
-    .delete()
-    .eq("address", fromAddress)
-    .in("claimId", formattedClaimIds)
-    .select();
+  // Wait for transaction to be confirmed for 5 blocks
+  if (await tx.wait(5).then((receipt) => receipt.status === 1)) {
```

Comment:
Hmm... this will exacerbate the UX delay, since the claimable items will continue to show up as claimable until it gets deleted from supabase.

So You claim all fractions, it succeeds, and for 5 blocks you'll still see them as claimable.

I guess it's better to fix the bug, but just an FYI

Reply:
CC @ccerv1

Reply:
We can do the same in the FE. As long as the tx is being awaited, show a spinner on the "claim" button for example. Block times also differ per chain so this measure is probably most consistent

Reply:
fair enough, that works for me.

Reply:
Added pending state to tx hooks and use them to add spinners and disable buttons